packages/salesforcedx-vscode-lwc/src/commands/forceLightningLwcPreview.ts

```diff
@@ -72,31 +126,103 @@ export async function forceLightningLwcPreview(sourceUri: vscode.Uri) {
+  if (!desktopSelected) {
+    let placeholderText = nls.localize(
+      'force_lightning_lwc_preview_target_default'
+    );
+    const deviceConfig = getWorkspaceSettings().get('rememberDevice') || false;
+    const lastTarget = getRememberedDevice(platformSelection);
 
-  if (DevServerService.instance.isServerHandlerRegistered()) {
-    try {
-      await openBrowser(fullUrl);
-      telemetryService.sendCommandEvent(logName, startTime);
-    } catch (e) {
-      showError(e, logName, commandName);
+    // Remember device setting enabled and previous device retrieved.
+    if (deviceConfig && lastTarget) {
+      placeholderText = nls.localize(
+        'force_lightning_lwc_preview_target_remembered',
+        lastTarget
+      );
+      target = lastTarget;
     }
-  } else {
-    console.log(`${logName}: server was not running, starting...`);
-    const preconditionChecker = new SfdxWorkspaceChecker();
-    const parameterGatherer = new EmptyParametersGatherer();
-    const executor = new ForceLightningLwcStartExecutor({
-      openBrowser: true,
-      fullUrl
+    const targetName = await vscode.window.showInputBox({
+      placeHolder: placeholderText
     });
 
-    const commandlet = new SfdxCommandlet(
-      preconditionChecker,
-      parameterGatherer,
-      executor
-    );
+    if (targetName === undefined) {
+      vscode.window.showInformationMessage(
+        nls.localize('force_lightning_lwc_preview_device_cancelled')
+      );
+      return;
+    }
+
+    // New target device entered
+    if (targetName !== '') {
+      updateRememberedDevice(platformSelection, targetName);
+      target = targetName;
+    }
+
+    // Start Server if not running.
+    if (!DevServerService.instance.isServerHandlerRegistered()) {
+      console.log(`${logName}: server was not running, starting...`);
+      const preconditionChecker = new SfdxWorkspaceChecker();
+      const parameterGatherer = new EmptyParametersGatherer();
+      const executor = new ForceLightningLwcStartExecutor({
+        openBrowser: desktopSelected,
+        fullUrl
+      });
+
+      const commandlet = new SfdxCommandlet(
+        preconditionChecker,
+        parameterGatherer,
+        executor
+      );
+
+      await commandlet.run();
+      telemetryService.sendCommandEvent(logName, startTime);
+    } else if (desktopSelected) {
+      try {
+        await openBrowser(fullUrl);
+        telemetryService.sendCommandEvent(logName, startTime);
+      } catch (e) {
+        showError(e, logName, commandName);
+      }
+      return;
+    }
+
+    // Launch Mobile Device
+    const mobileCancellationTokenSource = new vscode.CancellationTokenSource();
+    const mobileCancellationToken = mobileCancellationTokenSource.token;
 
-    await commandlet.run();
-    telemetryService.sendCommandEvent(logName, startTime);
+    const command = new SfdxCommandBuilder()
+      .withDescription(commandName)
+      .withArg('force:lightning:lwc:preview')
```

Comment:
I.e. the logic started [here](https://github.com/brandonpage/salesforcedx-vscode/pull/1/files#diff-785bdec339e64f40207b4e9dcae14dd4R173) needs to be wrapped inside of this command's workflow.

Reply:
Which is all @trooper2013 's problem, so don't worry. 😉